variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -85,62 +76,91 @@ def fetch_bases(fasta, contig, length, start):
     return new_ref
 
 
-def calculate_new_alleles(old_ref, new_ref, old_alt, is_reverse_strand):
-    """Given allele and strand information, calculate the new (ref, alt) pair."""
-    # If reverse strand -> calculate complement
-    old_ref_processed = old_ref
-    old_alt_processed = old_alt
-    if is_reverse_strand:
-        old_ref_processed = Seq(old_ref_processed, generic_dna).reverse_complement()
-        old_alt_processed = Seq(old_alt_processed, generic_dna).reverse_complement()
-
-    # No changes
-    if old_ref_processed == new_ref:
-        return new_ref, old_alt_processed
-    # Ref and Alt are the same -> Alt changed to old Ref
-    if new_ref == old_alt_processed:
-        return new_ref, old_ref_processed
+def group_reads(bam_file_path):
+    with pysam.AlignmentFile(bam_file_path, 'rb') as inbam:
+        current_read_name = None
+        primary_group = None
+        secondary_group = None
+        for read in inbam:
+            if read.query_name == current_read_name:
+                pass
+            else:
+                if current_read_name:
+                    yield primary_group, secondary_group
+                primary_group = []
+                secondary_group = []
+            if read.is_secondary:
+                secondary_group.append(read)
+            else:
+                primary_group.append(read)
+            current_read_name = read.query_name
+        yield primary_group, secondary_group
+
+
+def _order_reads(group):
+    """Order read and return the most 5' (smallest coordinates) first."""
+    read1, read2 = group
+    if read1.pos <= read2.pos:
+        return read1, read2
     else:
-        return new_ref, old_alt_processed
+        return read2, read1
```

Comment:
This would not work in a rare situation where the reads align in an incorrect orientation to start with (see comments for `calculate_new_variant_definition`)

Reply:
This function is simply to initialise the reads' position. It does not make any statement about their orientation.

Reply:
Oh, that's right, it's done _post_ filtering. Sorry, my mistake!